src/net.cpp

```diff
@@ -1773,6 +1773,14 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
 
         addrman.ResolveCollisions();
 
+        // Open this connection as block-relay-only if we're already at our
+        // full-relay capacity, but not yet at our block-relay peer limit.
+        // (It should not be possible for fFeeler to be set if we're not
+        // also at our block-relay peer limit, but check against that as
+        // well for sanity.)
+        bool block_relay_only = nOutboundBlockRelay < m_max_outbound_block_relay && !fFeeler && nOutboundFullRelay >= m_max_outbound_full_relay;
+        bool block_relay_tor_only = block_relay_only && nOutboundBlockRelay >= MAX_BLOCKS_ONLY_CONNECTIONS;
```

Comment:
This logic appears to allow more than 2 block_relay_tor_only connections (we'd start off with 2 tor, but then if other block-relay peers are disconnected one by one, we'd only look to replace them with tor connections).  That seems like unintended behavior?